src/webui/api/torrentscontroller.cpp

```diff
@@ -875,13 +875,30 @@ void TorrentsController::removeTrackersAction()
 {
     requireParams({u"hash"_s, u"urls"_s});
 
-    const auto id = BitTorrent::TorrentID::fromString(params()[u"hash"_s]);
-    BitTorrent::Torrent *const torrent = BitTorrent::Session::instance()->getTorrent(id);
-    if (!torrent)
-        throw APIError(APIErrorType::NotFound);
+    const QString hashParam = params()[u"hash"_s];
+    const QStringList urlsParam = params()[u"urls"_s].split(u'|', Qt::SkipEmptyParts);
+
+    QStringList urls;
+    urls.reserve(urlsParam.size());
+    for (const QString &urlStr : urlsParam)
+        urls << QUrl::fromPercentEncoding(urlStr.toLatin1());
 
-    const QStringList urls = params()[u"urls"_s].split(u'|');
-    torrent->removeTrackers(urls);
+    // remove trackers from all torrents
+    if (hashParam == u"*"_s)
+    {
+        for (BitTorrent::Torrent *const torrent : asConst(BitTorrent::Session::instance()->torrents()))
+            torrent->removeTrackers(urls);
+    }
+    // remove trackers from specified torrent
+    else
+    {
+        const auto id = BitTorrent::TorrentID::fromString(hashParam);
+        BitTorrent::Torrent *const torrent = BitTorrent::Session::instance()->getTorrent(id);
+        if (!torrent)
+            throw APIError(APIErrorType::NotFound);
+
+        torrent->removeTrackers(urls);
+    }
```

Comment:
Maybe just me, I would suggest really follow the idea that the 'remove trackers' code path is shared. And only differ when processing the `hash` parameter.

Also, I suppose the `hash` should be able to take a list of comma separated hashes. Can be done in another PR.

nit: a comment line that separates the if-else cases breaks the flow of reading for me. I won't recommended that.

```suggestion
    QList<BitTorrent::Torrent *> torrents;
    if (hashParam == u"*"_s)  // remove trackers from all torrents
    {
        torrents = BitTorrent::Session::instance()->torrents();
    }
    else  // remove trackers from specified torrent
    {
        const auto id = BitTorrent::TorrentID::fromString(hashParam);
        BitTorrent::Torrent *const torrent = BitTorrent::Session::instance()->getTorrent(id);
        if (!torrent)
            throw APIError(APIErrorType::NotFound);
        torrents.append(torrent);
    }

    for (BitTorrent::Torrent *const torrent : asConst(torrents))
            torrent->removeTrackers(urls);
```

Reply:
Adjusted the logic to take this approach 👍